pkg/trace/api/api.go

```diff
@@ -427,12 +427,10 @@ func decodeTracerPayload(v Version, req *http.Request, ts *info.TagStats) (tp *p
 			return nil, false, err
 		}
 		var tracerPayload pb.TracerPayload
-		err = tracerPayload.Unmarshal(buf.Bytes())
+		_, err = tracerPayload.UnmarshalMsg(buf.Bytes())
```

Comment:
This was an unfortunate mistake. `Unmarshal()` is proto decoder, and `UnmarshalMsg()` is msgp decoder.

Reply:
This is crazy. We need an integration test for this or something that would've caught the problem. Or, is your newly added test the one that did catch it?

Reply:
Newly added test caught it.